sklearn/utils/validation.py

```diff
@@ -363,8 +363,10 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
     else:
         if ensure_2d:
             array = np.atleast_2d(array)
-
-        array = np.array(array, dtype=dtype, order=order, copy=copy)
+        if not copy:
```

Comment:
> > -        array = np.array(array, dtype=dtype, order=order, copy=copy)
> > +        if not copy:

> See arthurmensch@976e28e#commitcomment-11837796.

> If copy is not asked specifically, we should avoid it : np.array does make a
> copy of a memmap input whatever the argument copy

OK, you should add a comment about this: it's not obvious at all from the
code that this line is there for memmap.